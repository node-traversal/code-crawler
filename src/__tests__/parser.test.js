const _ = require('lodash');
const fs = require('fs');
const path = require('path');
const parser = require('../parser');

const basePath = process.cwd();
const examplesPath = path.join(basePath, 'examples/components');

describe('The parser should', () => {
    const options = {
        basePath: examplesPath,
        packageName: '@my-project/the-package'
    };

    const parse = (file) => {
        const filePath = path.join(examplesPath, file);
        const code = fs.readFileSync(filePath, 'utf-8');
        return parser.parserReactCode(code, _.assign({}, options, { filePath }));
    };

    it('parse export with constant instead of import', () => {
        expect(parse('index-exports-default.js')).toMatchSnapshot();
    });

    it('parse re-export with star import', () => {
        expect(parse('index-re-export-star.js')).toMatchSnapshot();
    });

    it('parse re-export with aliased import', () => {
        expect(parse('index-re-export-alias.js')).toMatchSnapshot();
    });

    it('parse aliased default import', () => {
        expect(parse('import-named-default.js')).toMatchSnapshot();
    });
});
