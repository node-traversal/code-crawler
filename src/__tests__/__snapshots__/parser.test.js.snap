// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The parser should parse aliased default import 1`] = `
Object {
  "deps": Array [
    Object {
      "name": "react",
      "var": "React",
    },
    Object {
      "name": "prop-types",
      "var": "PropTypes",
    },
  ],
}
`;

exports[`The parser should parse export with constant instead of import 1`] = `
Object {
  "deps": Array [
    Object {
      "name": "@my-project/the-package/TreeView",
      "var": "default",
    },
  ],
}
`;

exports[`The parser should parse re-export with aliased import 1`] = `
Object {
  "deps": Array [
    Object {
      "name": "@my-project/the-package/foobar",
      "var": "foo",
    },
    Object {
      "name": "@my-project/the-package/foobar",
      "var": "bar",
    },
  ],
}
`;

exports[`The parser should parse re-export with star import 1`] = `
Object {
  "deps": Array [
    Object {
      "name": "@my-project/the-package/other",
    },
  ],
}
`;
